docs/zkasm/test-infra.md

```diff
@@ -0,0 +1,98 @@
+# About
+
+We want to build a robust testing infrastructure which will:
+
+- allow easily add new test.
+- allow to reuse existing cranelift tests.
+- be easy to introduce new type of tests.
+
+We want keep our testing infra flexible and friendly to user. Also, this version is not final, and will be improved. For example, testing the interpreter by running wasm in it is out of scope for now.
+
+
+# Testing use cases
+
+We have two main use cases for our testing infra:
+
+- CI. We want run our tests in CI to keep quality of code in main branch on high level
+- Developing. We want our tests to be easy to run by developer in local machine and see how current version of code works. This use case give us a new requirement -- our testing infra should be fast enough, ideally run in few seconds.
+
+# How to use testing infra
+
+## How test file looks like?
```

Comment:
I'm not sure it is worth to reiterate what is effectively the contents of cranelift/filetests/README.md. Just link to that instead.

I would also perhaps consider contributing improvements to that and related files directly upstream (so long as they don't have much specific to zkasm.) I'm sure they'd appreciate more thorough docs.

Reply:
Hmm, don't think I wrote something better that cranelift docs have, but I replaced it by link to docs, thanks